### PR TITLE
Feature/load single slice

### PR DESCRIPTION
**Pull request by frasercl**
Resolve (?) #110: change relevant loading items (`OMEZarrLoader`, `IVolumeLoader`, `LoadSpec`) to allow loading of subregions from zarr arrays, including single slices.

### Changes:

- `LoadSpec` subregion fields (`minx`, `maxx`, `miny`, etc.) may be undefined to indicate loading the max extent of that axis 
- `OMEZarrLoader` now pays attention to these fields
- `FetchZarrWorker` now uses the `zarr.getRaw` method, which removes the need to de-nest typed arrays in the worker
- `IVolumeLoader` allows passing in an optional explicit `LoadSpec` on data load which overrides the one in the target `Volume`

### Does NOT (yet) change:

- Other discussed changes to `LoadSpec`: replace `subpath` with `scaleLevel`/`multiscale`, add array of channel indexes
- Replace the per-channel callback API with one based on an array of promises
- Make `Volume` aware that it contains a subset of its full extent. Currently, `OMEZarrLoader` will dutifully load a smaller-than-expected array of channel data into the full dimensions of the volume, and the volume will show up but with a lot of blank space. It's unclear how `Volume` should handle this.
- Allow loading original data types, rather than converting to uint8 (#121), though the simplifications to `FetchZarrWorker` will make this easier

### CODE REVIEW

File: src/test/RequestQueue.test.ts [hunk 1]
```diff
@@ -329,7 +333,7 @@ describe("test RequestQueue", () => {
       expect(count).to.equal(0);
     });
 
-    async function mockLoader(loadSpec: LoadSpec, maxDelayMs = 10.0): Promise<TypedArray> {
+    async function mockLoader(loadSpec: Required<LoadSpec>, maxDelayMs = 10.0): Promise<TypedArray> {
```
Comment (toloudis) [inline src/test/RequestQueue.test.ts:336]: TL;DG: what does Required mean ?

File: src/workers/FetchZarrWorker.ts [hunk 2]
```diff
@@ -1,83 +1,61 @@
...
+  channelData.forEach((val: number, idx: number) => {
+    u8[idx] = ((val - min) / range) * 255;
+  });
+
   return u8;
 }
 
-self.onmessage = function (e) {
-  const time = e.data.time;
+self.onmessage = async (e: MessageEvent<FetchZarrMessage>) => {
```
Comment (toloudis) [inline src/workers/FetchZarrWorker.ts:36]: good use of typing for this

**Review by toloudis — APPROVED** [review]
looks great apart from one thing to check with the `minx || null` thing

**Review by ShrimpCryptid — APPROVED** [review]
Looks good, no questions beyond what Dan's brought up already

**Pull request merged**